src/main/java/mate/academy/rickandmorty/service/DataDownLoadService.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterResponseDto;
+import mate.academy.rickandmorty.exception.DataProcessException;
+import mate.academy.rickandmorty.mapper.Mapper;
+import mate.academy.rickandmorty.model.Character;
+import mate.academy.rickandmorty.repository.Repository;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class DataDownLoadService {
+    private static final String URL = "https://rickandmortyapi.com/api/character/";
+    private static final int NUMBER_OF_CHARACTERS = 20;
+    private final Mapper mapper;
+    private final ObjectMapper jasonMapper;
+    private final Repository repository;
+
+    public void getAllCharactersFromApi() {
+        HttpClient client = HttpClient.newHttpClient();
```

Comment:
1 You instantiate the client after each method invocation... Fix it. 

2. Consider using the spring style (create a bean and inject it throughout your app)
```suggestion
        HttpClient client = HttpClient.newHttpClient();
```